src/megatron/bridge/training/train.py

```diff
@@ -319,7 +320,7 @@ def train(
                     model_config.param_sync_func = param_sync_func
                     pre_hook_enabled = True
                     # Set the manual hooks when CUDA Graphs are used.
-                    if model_config.external_cuda_graph:
+                    if model_config.cuda_graph_impl == "transformer_engine":
```

Comment:
there also needs to be handling for the skipped iterations in L280 in case there's overlap with the graph capture steps